### ISSUE TITLE
More accurate upload progress

### ISSUE THREAD

**Issue by mariusrak** [VeliovGroup/Meteor-Files#704]
Hi, would it be possible to provide bytes uploaded instead of percentage in on progress callback? There's no problem for anybody to calculate percentage but it is impossible to get accurate bytes uploaded from percentage.
Thanks.

**Comment by jankapunkt** [comment]
I will check it out and let you know if it's feasible to implement without causing further trouble.

**Comment by jankapunkt** [comment]
Okay currently the progress is directly set as event parameter [in the code](https://github.com/VeliovGroup/Meteor-Files/blob/dev/upload.js#L269):

```javascript
      this.addListener('calculateStats', helpers.throttle(() => {
        const _t = (this.transferTime / this.sentChunks) / this.config.streams;
        this.result.estimateTime.set((_t * (this.fileLength - this.sentChunks)));
        this.result.estimateSpeed.set((this.config.chunkSize / (_t / 1000)));

        const progress = Math.round((this.sentChunks / this.fileLength) * 100);
        this.result.progress.set(progress);
        this.config.onProgress && this.config.onProgress.call(this.result, progress, this.fileData);
        this.result.emit('progress', progress, this.fileData);
      }, 250));
```

A possible implementation, that also ensures backwards compatibility, could emit with an additional parameter:

```javascript
this.result.emit('progress', progress, this.fileData, { sentChunks: this.sentChunks, maxChunks: this.fileLength });
```

@dr-dimitru do you see any issues with this?

**Comment by mariusrak** [comment]
Maybe would be better to add new field e.g. `sentChunks` and not touch the
progress field at all.

**Comment by jankapunkt** [comment]
@mariusrak please look at my proposal at the end of my last comment. There I add a new field `sentChunks` at the end of the params list of the `progress` event without altering the original `progress` behavior. So we just add as few code as possible and also have the chunks added.

**Comment by jankapunkt** [comment]
I added a new PR #705

**Comment by mariusrak** [comment]
I've just realized, I'm accessing the progress not from event but through upload instance, where it is reactive var. Would this PR affect also my case and will I be able to receive the information about sent chunks? Thanks.

**Comment by jankapunkt** [comment]
Actually not but it should not be too hard to add another reactive variable `uploadInstance.sentChunks`.

**Comment by dr-dimitru** [comment]
Hey @mariusrak 👋

Please, see our review in #707, __action pending__
//cc @jankapunkt

**Comment by dr-dimitru** [comment]
Published as [`v1.13.0`](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.13.0). 
Feel free to close it in case if the issue is solved on your end.
